### DISCUSSION TITLE
Directus stops when SSH connection is terminated

### DISCUSSION THREAD

**Discussion by thammarith** [directus/directus#13602] [Archive]
I installed Directus ([following this doc](https://docs.directus.io/self-hosted/installation/cli)) on AWS Lightsail. I ssh-ed my machine and ran `npx directus start`. Everything works fine until the SSH connection timed out and got disconnected. Now I cannot access Directus. I have to SSH back in and run `npx directus start` again. 

What do I have to do to run it forever so it's like a normal website without having to SSH in and run `npx directus start` every time?

**Comment by rijkvanzanten** [comment]
While I'm not familiar with lightsail, it sounds like you might want to run the Directus process using something like pm2 to keep it alive

**Reply by thammarith** [reply]
I hate to have yet another npm package to solve this but it seems like pm2 is really the only easiest way to do. Thanks!

**Reply by rijkvanzanten** [reply]
You don't _have_ to. You could also use something like `screen` or `tmux`, or rely on another service to run the node process like through a container in Docker, etc. That being said, you'd want to rely on something that will keep the process alive as well, rather than just run the process and allow you to disconnect (like `screen` would be). Something like `pm2` or kubernetes will make sure to restart the process if Node errors out unexpectedly 👍🏻

**Comment by adnanf2** [comment]
I was having the same issue on DreamHost. The workaround I found was to use PUTTY to start a new SSH session and keep it alive. Under the connection tab add 240 seconds between keepalives and check the enable TCP keepalive option. Then connect to your server and start Directus. Hope it helps.

**Comment by br41nslug** [comment]
As Rijk suggested using a daemon that will keep the process running independently of the SSH session is the way to go here. PM2 is really build for this (https://pm2.keymetrics.io/docs/usage/process-management/) but general linux tools like `screen` works wonders too but may not be the best for production environments (https://linuxize.com/post/how-to-use-linux-screen/)
